File: Unity/TSProj/src/mods/ModTheSpire/Scripts/Power/RitualPower.ts
import { ApplyPowerAction } from "../Action/Common/ApplyPowerAction";
import { AtEndOfTurnMessage } from "../Events/AtEndOfTurnMessage";
import FileHelper from "../FileHelper";
import { LocalizationProperty } from "../Gen/DB/Localization";
import { StringHelper } from "../StringHelper";
import { AbstractPower, PowerType } from "./AbstractPower";
import { StrengthPower } from "./StrengthPower";

export class RitualPower extends AbstractPower{
    Icon: string = FileHelper.FormatPath("Powers/ritual.png");
    Id: string = "Ritual";
    Name: string = LocalizationProperty.Read("能力-仪式");

    get Type(): PowerType {
        return PowerType.Buff;
    }

    public constructor(private skipFirst:boolean) {
        super();
    }
    
    GetDescription(...args: any[]): string {
        return StringHelper.FormatColorString(LocalizationProperty.Read("能力-仪式描述")).format(this.Amount);
    }

    protected AtEndOfTurn(msg: AtEndOfTurnMessage) {
        super.AtEndOfTurn(msg);
        if(this.skipFirst){
            this.skipFirst = false;
            return;
        }
        this.Flash();
        this.AddToBot(new ApplyPowerAction(this.Owner,this.Owner,new StrengthPower(),this.Amount));
    }
}

File: Unity/Assets/Scripts/ModTheSpire/Power/Buff/RitualPower.cs
using ModTheSpire.Effect.Other;
using ModTheSpire.Event.Ids;
using TCG.Config;

namespace ModTheSpire.Power.Buff
{
    public class RitualPower : AbstractPower
    {
        public override string Id { get; } = nameof(RitualPower);
        public override string Name { get; } = LocalizationProperty.Read("能力-仪式");
        public override string Icon { get; } = "Powers/ritual.png";
        public override PowerType Type { get; } = PowerType.Buff;
        private bool SkipFirst { get; set; } = true;
        public override string GetDescription(object[] args)
        {
            return string.Format(LocalizationProperty.Read("能力-仪式描述"), this.Amount);
        }

        protected override void AtEndOfTurn(AtEndOfTurnMessage msg)
        {
            base.AtEndOfTurn(msg);
            if(this.SkipFirst){
                this.SkipFirst = false;
                return;
            }
            this.Flash();
            this.AddToBot(new ApplyPowerAction(this.Owner,this.Owner,new StrengthPower(),this.Amount));
        }
    }
}